### PR TITLE
[Besu] Fixed redeploying network while network is running

### PR DESCRIPTION
**Pull request by azoumi**
Genesis block is now saved in the vault, which should fix the bug completely

Signed-off-by: azoumi <hafid.azoumi@accenture.com>

**Changelog**
- Besu genesis file is now saved in the vault
- Genesis file will get retreived from the vault if not available locally.
- Tested with redeploying same network while network is running
- Tested with adding new organization

 
**Reviewed by**
@lakshyakumar 

 

**Linked issue**
#1034

### CODE REVIEW

File: platforms/hyperledger-besu/configuration/roles/create/crypto/ibft/tasks/add_to_vault.yaml [hunk 1]
```diff
@@ -8,3 +8,11 @@
...
   loop_control:
     loop_var: peer
+
+- name: Copy genesis to Vault
+  shell: |
+    vault kv put secret/{{ component_ns }}/crypto/genesis genesis="{{ genesis }}" 
+  environment:
+    VAULT_ADDR: "{{ vault.url }}"
+    VAULT_TOKEN: "{{ vault.root_token }}"
+  when: add_new_org is not defined or add_new_org|bool == false
```
Comment (lakshyakumar) [inline platforms/hyperledger-besu/configuration/roles/create/crypto/ibft/tasks/add_to_vault.yaml:18]: add a newline

File: platforms/hyperledger-besu/configuration/roles/create/ibft/tasks/main.yaml [hunk 2]
```diff
@@ -33,6 +33,30 @@
...
+  ignore_errors: yes
+  when: stat_result.stat.exists == False
+
+#This task only runs when there is no local genesis file
+- name: Copy genesis from vault to correct path
+  shell: |
+    echo {{ vault_genesis.stdout }} > {{ network.config.genesis }}
+  when: stat_result.stat.exists == False
+
+
```
Comment (lakshyakumar) [inline platforms/hyperledger-besu/configuration/roles/create/ibft/tasks/main.yaml:59]: remove extra empty lines

**Review by lakshyakumar — COMMENTED** [review]
Is the code is tested with the addition of new member node in besu?

**Review by lakshyakumar — CHANGES_REQUESTED** [review]
Genesis file should be deleted from vault using the reset network playbook.

**Comment by azoumi** [comment]
> 
> 
> Is the addition of new member org is tested after the changes?

Yes that is tested.

**Comment by azoumi** [comment]
> 
> 
> Genesis file should be deleted from vault using the reset network playbook.

Added.

File: platforms/hyperledger-besu/configuration/roles/create/crypto/ibft/tasks/add_to_vault.yaml (hunk 1 above)
Comment (azoumi) [reply platforms/hyperledger-besu/configuration/roles/create/crypto/ibft/tasks/add_to_vault.yaml]: Fixed

File: platforms/hyperledger-besu/configuration/roles/create/ibft/tasks/main.yaml (hunk 2 above)
Comment (azoumi) [reply platforms/hyperledger-besu/configuration/roles/create/ibft/tasks/main.yaml:59]: Fixed

**Review by lakshyakumar — APPROVED** [review]

**Review by sownak — APPROVED** [review]

**Pull request merged**